python/pyarrow/tests/test_pandas.py

```diff
@@ -2164,9 +2172,17 @@ def test_nested_large_list(self):
         s = (pa.array([[[1, 2, 3], [4]], None],
                       type=pa.large_list(pa.large_list(pa.int64())))
              .to_pandas())
-        tm.assert_series_equal(
-            s, pd.Series([[[1, 2, 3], [4]], None], dtype=object),
-            check_names=False)
+
+        # pandas.testing generates a
+        # DeprecationWarning: elementwise comparison failed
+        # numpy.VisibleDeprecationWarning: Creating an ndarray
+        #     from ragged nested sequences ...
```

Comment:
These warnings are happening inside of the `pandas.testing` module though.  Unless I'm misunderstanding.. should we open an issue with pandas then?

Reply:
Ah! Yes, perhaps open an issue with them, in case there's a workaround?

Reply:
Hmm, actually, when making an example for the issue to report, I _think_ it maybe how our implementation of converting a variable list array to pandas:

```python
import pyarrow as pa
import pandas as pd
import pandas.testing as tm

s = (pa.array([[[1, 2, 3], [4]], None],
                      type=pa.large_list(pa.large_list(pa.int64())))
             .to_pandas())

# This will raise the warning
tm.assert_series_equal(
     s, pd.Series([[[1, 2, 3], [4]], None], dtype=object),
     check_names=False)

# but so will this:
s.equals(pd.Series([[[1, 2, 3], [4]], None], dtype=object))

# this will not
tm.assert_series_equal(
     pd.Series([[[1, 2, 3], [4]], None], dtype=object),
     pd.Series([[[1, 2, 3], [4]], None], dtype=object),
     check_names=False)
```

Therefore I think a new issue to be made to correct this on our end after all. :thinking:

Reply:
For now, I've made issues 
- https://github.com/apache/arrow/issues/14759 
- https://github.com/apache/arrow/issues/14760. 

I can create separate PRs for them I think if that's agreeable.

Reply:
Yes, separate PRs sound fine.

Reply:
See my answer in https://github.com/apache/arrow/issues/14759, for the warning coming from pandas' testing, I think we can ignore that warning on our side (and maybe report it to pandas)

Reply:
Thanks @jorisvandenbossche! Okay, I'll add back the ignoring of this error then here, and also removed saying this closed [ARROW-17651](https://issues.apache.org/jira/browse/ARROW-17651)